[code-infra] Setup error message minification (#1463)
mui/base-ui: docs/src/app/(public)/(content)/production-error/[code]/PageContent.mdx

> # Production error

Reviewer: It feels like we should change this to improve the DX:

- People arriving here would see stuff like this in their console:

<img width="523" height="76" alt="SCR-20250906-mzzx" src="https://github.com/user-attachments/assets/bd43706b-dbad-4e7e-8f13-2a80d09f5acf" />

- If we create 73 pages with the same H1 and title, that looks like duplication of content; it feels like it defeats the point of slowing down the build time.

So I would expect to see the error code somewhere, like it does in https://react.dev/errors/130. It's in the h1, it doesn't have to be there, but one option.

Reviewer: **Off-topic**. Speaking of the error DX, I wonder about this pattern https://nextjs.org/docs/messages/missing-suspense-with-csr-bailout. But it feels a lot like duplication of content. Meaning this should be in the docs. For example `missingSuspenseWithCSRBailout` is only documented in the error page 🙈. So it's probably a no-no, the links in the errors should push back to the docs, and the docs needs to be clear, so the error message https://www.notion.so/mui-org/Technical-writing-style-guide-7e55b517ac2e489a9ddb6d0f6dd765de#85219822c3194b6f8a49ee08ea82b90a.

{/**
@typedef Props
@property {string} msg The raw error message.
**/}

<Subtitle>Explanation for minified production error message.</Subtitle>
<Meta
  name="description"
  content="In the production build, error messages are minified to keep your application lightweight."
/>

A minified Base UI error occurred in the production build of React.

The full error message:

> <ErrorDisplay msg={props.msg} />

Reviewer: Current https://deploy-preview-1463--base-ui.netlify.app/production-error/1

<img width="544" height="234" alt="SCR-20250906-ndbn" src="https://github.com/user-attachments/assets/e715539b-9a9e-47ff-9608-6b2e6b30f591" />

I doubt this is how this should look. It feels like we should render the error red `#ff0000`, like people will see the error in their console first. This feels better: https://react.dev/errors/130.

Reviewer 2: The design can be iterated on by us as well as the Base UI team. The PR is more about setting up the infrastructure for error codes.

> For debugging, we recommend using the development build, which provides additional warnings about potential issues. If an exception occurs in the production build, this page will reconstruct the original error message.
> 
> The original error you encountered:

Reviewer 3: I think this can be a little bit simpler:
```suggestion
A Base UI error occurred in the production build of React. It's recommended to use the development build for debugging as it provides additional warnings about potential issues.
```
